community/index/src/main/java/org/neo4j/index/internal/gbptree/GBPTree.java

```diff
@@ -523,7 +529,7 @@ private static void writerHeader( PagedFile pagedFile, Header.Writer headerWrite
         // Write/carry over header
         int headerOffset = cursor.getOffset();
         int headerDataOffset = headerOffset + Integer.BYTES; // will contain length of written header data (below)
-        if ( otherState.isValid() )
+        if ( otherState.isValid() || headerWriter != CARRY_OVER_PREVIOUS_HEADER )
```

Comment:
Because if we use `CARRY_OVER_PREVIOUS_HEADER` that means we copy the header from `otherState`. But if `otherState` is not valid then we will write corrupt data to our tree state, which would be sad. However, if the `headerWriter` is _not_ a `CARRY_OVER_PREVIOUS_HEADER` it is safe for us to just write.